src/status_im2/contexts/wallet/events.cljs

```diff
@@ -321,3 +321,24 @@
 (rf/reg-event-fx :wallet/select-send-address
  (fn [{:keys [db]} [address]]
    {:db (assoc db :wallet/send-address address)}))
+
+(rf/reg-event-fx :wallet/get-address-details-success
+ (fn [{:keys [db]} [{:keys [hasActivity]}]]
+   {:db (assoc-in db
+         [:wallet :ui :watch-address-activity-state]
+         (if hasActivity :has-activity :no-activity))}))
+
+(rf/reg-event-fx :wallet/clear-address-activity-check
+ (fn [{:keys [db]}]
+   {:db (update-in db [:wallet :ui] dissoc :watch-address-activity-state)}))
+
+(rf/reg-event-fx :wallet/get-address-details
+ (fn [{:keys [db]} [address]]
+   {:db (assoc-in db [:wallet :ui :watch-address-activity-state] :scanning)
+    :fx [[:json-rpc/call
+          [{:method     "wallet_getAddressDetails"
+            :params     [(chain/chain-id db) address]
```

Comment:
Seems a little strange that we have to pass the whole `db` to get a `chain-id`.

Also that chain-id function is not clear, I think it should be named something like `get-current-network-chain-id` (longer but clearer). I know you didn't addded this function, and we have like 22 usages. since we are rebuilding the wallet  events, it may worth considering renaming.

wdyt?

Reply:
I totally agree that it's not necessary to pass the whole DB to get one value out of a map.

The `chain/chain-id` method uses the `:networks/current-network` to get the network id from `:networks/:networks`.

<img width="392" src="https://github.com/status-im/status-mobile/assets/19339952/0d8aa11a-28f8-4971-9022-30d5aedb2289">

I'm thinking of storing the current network ID in the root of app-db, just like we store the network name (`:networks/current-network`). So, we can use it directly whenever we need it instead of using these helper functions.

Reply:
> I'm thinking of storing the current network ID in the root of app-db,

:thinking:  I think it's already stored in the root app-db, since the function:

```clojure
(defn get-current-network
  [m]
  (get (:networks/networks m) (:networks/current-network m)))
```
extracts it from `m` which actually is the whole db.

Is there a better place for it to store it? if you think root is the best, then let's keep it there :+1: 

I'm not agains using a helper function, just passing the whole db is like too much IMO.

also that function is being used in other places in `status-im` namespaces, the only usages in `status-im2` are in our wallet implementation:

![image](https://github.com/status-im/status-mobile/assets/90291778/a6780c95-4c95-4de8-ad40-b7ccd91c9db3)

So I think it's a good point to improve :+1:

Reply:
Btw that sub is old data afaik, let's use wallet/network-details for network related details